contracts/0.8.25/upgrade/V3Template.sol

```diff
@@ -0,0 +1,373 @@
+}
+
+interface IOracleReportSanityChecker is IAccessControlEnumerable {
+    function ALL_LIMITS_MANAGER_ROLE() external view returns (bytes32);
+    function EXITED_VALIDATORS_PER_DAY_LIMIT_MANAGER_ROLE() external view returns (bytes32);
+    function APPEARED_VALIDATORS_PER_DAY_LIMIT_MANAGER_ROLE() external view returns (bytes32);
+    function ANNUAL_BALANCE_INCREASE_LIMIT_MANAGER_ROLE() external view returns (bytes32);
+    function SHARE_RATE_DEVIATION_LIMIT_MANAGER_ROLE() external view returns (bytes32);
+    function MAX_VALIDATOR_EXIT_REQUESTS_PER_REPORT_ROLE() external view returns (bytes32);
+    function MAX_ITEMS_PER_EXTRA_DATA_TRANSACTION_ROLE() external view returns (bytes32);
+    function MAX_NODE_OPERATORS_PER_EXTRA_DATA_ITEM_ROLE() external view returns (bytes32);
+    function REQUEST_TIMESTAMP_MARGIN_MANAGER_ROLE() external view returns (bytes32);
+    function MAX_POSITIVE_TOKEN_REBASE_MANAGER_ROLE() external view returns (bytes32);
+    function SECOND_OPINION_MANAGER_ROLE() external view returns (bytes32);
+    function INITIAL_SLASHING_AND_PENALTIES_MANAGER_ROLE() external view returns (bytes32);
+}
+
+
+/**
+* @title Lido V3 Upgrade Template
+*
+* @dev Must be used by means of two calls:
+*   - `startUpgrade()` after upgrading LidoLocator and before everything else
+*   - `finishUpgrade()` as the last step of the upgrade
+*/
+contract V3Template is V3Addresses {
+    //
+    // Events
+    //
+
+    event UpgradeFinished();
+
+    //
+    // -------- Constants --------
+    //
+
+    uint256 internal constant ACCOUNTING_ORACLE_CONSENSUS_VERSION = 4;
+    uint256 internal constant EXPECTED_FINAL_LIDO_VERSION = 3;
+    uint256 internal constant EXPECTED_FINAL_ACCOUNTING_ORACLE_VERSION = 3;
+
+    bytes32 public constant DEFAULT_ADMIN_ROLE = 0x00;
+
+    // Timestamp since startUpgrade() and finishUpgrade() revert with Expired()
+    // This behavior is introduced to disarm the template if the upgrade voting creation or enactment didn't
+    // happen in proper time period
+    uint256 public constant EXPIRE_SINCE_INCLUSIVE = 1761868800; // 2025-10-31 00:00:00 UTC
+
+    // Initial value of upgradeBlockNumber storage variable
+    uint256 internal constant UPGRADE_NOT_STARTED = 0;
+
+    uint256 internal constant INFINITE_ALLOWANCE = type(uint256).max;
+
+    //
+    // Structured storage
+    //
+
+    uint256 public upgradeBlockNumber = UPGRADE_NOT_STARTED;
+    bool public isUpgradeFinished;
+    uint256 internal initialOldBurnerStethSharesBalance;
+    uint256 internal initialTotalShares;
+    uint256 internal initialTotalPooledEther;
+
+
+    /// @param _params Params required to initialize the addresses contract
+    constructor(V3AddressesParams memory _params) V3Addresses(_params) {
+    }
+
+    /// @notice Must be called after LidoLocator is upgraded
+    function startUpgrade() external {
+        if (msg.sender != AGENT) revert OnlyVotingCanUpgrade();
```

Comment:
So Agent or Voting? :thinking:

Reply:
set agent for now, easy to change in case